fix: TFKerasTrial on tf2 with tf.compat.v1.disable_v2_behavior.

ioga opened the pull request:
## Description

<!---
Lead with the intended commit body in this description field. For breaking changes,
please include "BREAKING CHANGE:" at the beginning of your commit body.
At minimum, this section should include a bracketed reference to the Jira ticket,
e.g. "[DET-1234]". When squash-and-merging, copy this directly into the description field.
-->


## Test Plan

<!---
Describe the situations in which you've tested your change, and/or a screenshot as appropriate.
Reviewers may ask questions of this test plan to ensure adequate manual coverage of changes.
-->


## Commentary (optional)

<!---
Use this section of your description to add context to the PR. Could be for particularly
tricky bits of code that could use extra scrutiny, historical context useful for reviewers, etc.
You may intentionally leave this section blank and remove the title.
--->


## Checklist

- [ ] User-facing API changes need the "User-facing API Change" label.
- [ ] Release notes should be added as a separate file under `docs/release-notes/`.
See [Release Note](../docs/release-notes/README.md) for details.
- [ ] Licenses should be included for new code which was copied and/or modified from any external code.


<!---
## Title

Example title: "docs: tweak recommended "pip install" usage".

Specifically, this title should contain a type and a description
of the change being made:

User-facing change types:
- docs: docs-only change
- feat: new user-facing feature
- fix: bug fix
- perf: performance improvement

Internal change types:
- build: build system change (anything in a `Makefile`, mostly)
- chore: any internal change not covered by another type
- ci: anything that touches `.circleci`
- refactor: internal refactor
- style: style change
- test: new tests

See https://www.conventionalcommits.org/en/v1.0.0/ for background.

The first line should also:
- be at most 89 characters long
- contain a description that is at most 72 characters long
- not end with sentence-ending punctuation
- start (after the type) with a lowercase imperative ("add", "fix")
-->


[DET-1234]: https://determinedai.atlassian.net/browse/DET-1234

rb-determined-ai commented on harness/determined/keras/_tf_keras_trial.py:
```diff
@@ -567,7 +567,7 @@ def _save_checkpoint(self, path: pathlib.Path) -> workload.Response:
         # Save RNG state.
         rng_state = {"np_rng_state": np.random.get_state(), "random_rng_state": random.getstate()}
-        if version.parse(tf.__version__) < version.parse("2.0.0"):
+        if not (version.parse(tf.__version__) >= version.parse("2.0.0") and tf._tf2.enabled()):
```
Why convert the  `<` into a `not >=`, out of curiosity?

ioga replied:
it may be matter of taste, but in my opinion, `NOT(version >= 2 AND version2-specific-condition)` is more readable, because inside of NOT you have a standard pattern of checking for something specific for the new version. 
This makes use of short circuiting (otherwise `tf._tf2` will error out), and short circuiting based on OR's may be harder to read fast.

rb-determined-ai reviewed (commented):
What about the other cases where `version.parse()` is used in out `keras` module?

rb-determined-ai commented:
What about the other cases where version.parse() is used in out keras module?

ioga commented:
I didn't change other cases of version.parse() cause it's not clear they need this particular check based only on the code and in the absence particular bugs. 
But I've discovered that the RNG state management in keras and estimator was done inconsistently, so I moved it out for reuse, started to check for eager execution because it matters, and added a test for this bug.

rb-determined-ai reviewed (approved):
Looks great.  Tried to break it but couldn't.
